pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,433 @@
+	restConfig, err := e.getKubeConfig(cluster)
+	if err != nil {
+		return cluster, fmt.Errorf("error getting service account token: %v", err)
+	}
+
+	clusterDialer, err := e.ClientDialer.ClusterDialer(cluster.Name)
+	if err != nil {
+		return cluster, err
+	}
+
+	saToken, err := generateSAToken(restConfig, clusterDialer)
+	if err != nil {
+		return cluster, fmt.Errorf("error getting service account token: %v", err)
+	}
+
+	cluster = cluster.DeepCopy()
+	cluster.Status.ServiceAccountToken = saToken
+	return e.ClusterClient.Update(cluster)
+}
+
+// buildAKSCCCreateObject returns an object that can be used with the kubernetes dynamic client to
+// create an AKSClusterConfig that matches the spec contained in the cluster's AKSConfig.
+func buildAKSCCCreateObject(cluster *mgmtv3.Cluster) (*unstructured.Unstructured, error) {
+	aksClusterConfig := aksv1.AKSClusterConfig{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "AKSClusterConfig",
+			APIVersion: aksV1,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: cluster.Name,
+			OwnerReferences: []v1.OwnerReference{
+				{
+					Kind:       cluster.Kind,
+					APIVersion: rbac.RancherManagementAPIVersion,
+					Name:       cluster.Name,
+					UID:        cluster.UID,
+				},
+			},
+		},
+		Spec: *cluster.Spec.AKSConfig,
+	}
+
+	// convert AKS cluster config into unstructured object so it can be used with dynamic client
+	aksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&aksClusterConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &unstructured.Unstructured{
+		Object: aksClusterConfigMap,
+	}, nil
+}
+
+// recordAppliedSpec sets the cluster's current spec as its appliedSpec
+func (e *aksOperatorController) recordAppliedSpec(cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	if reflect.DeepEqual(cluster.Status.AppliedSpec.AKSConfig, cluster.Spec.AKSConfig) {
+		return cluster, nil
+	}
+
+	cluster = cluster.DeepCopy()
+	cluster.Status.AppliedSpec.AKSConfig = cluster.Spec.AKSConfig
+	return e.ClusterClient.Update(cluster)
+}
+
+// deployAKSOperator looks for the rancher-aks-operator app in the cattle-system namespace, if not found it is deployed.
+// If it is found but is outdated, the latest version is installed.
+func (e *aksOperatorController) deployAKSOperator() error {
+	return e.DeployOperator(aksOperator, aksOperatorTemplate, aksShortName)
+}
+
+func (e *aksOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, bool, error) {
+	var publicAccess bool
+
+	restConfig, err := e.getKubeConfig(cluster)
+
+	netDialer := net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+	serviceToken, err := generateSAToken(restConfig, netDialer.DialContext)
+	if err != nil {
+		var dnsError *net.DNSError
+		if stderrors.As(err, &dnsError) && !dnsError.IsTemporary {
+			return "", true, nil
+		}
+	} else {
+		publicAccess = false
+	}
```

Comment:
This line isn't necessary as publicAccess is already false if the block above wasn't entered.

Reply:
removed

Reply:
removed